File: javatest/omnia/data/cache/CachedDoubleTest.java
package omnia.data.cache;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import static org.junit.Assert.assertTrue;

@RunWith(JUnit4.class)
public class CachedDoubleTest {

  @Test
  public void cache_didReturnSimpleDoubleCache() {
    assertTrue(CachedDouble.cache(() -> 132.0) instanceof SimpleDoubleCacher);
  }
}

File: omnia/src/commonTest/omnia/data/cache/CachedDoubleTest.kt
package omnia.data.cache

import kotlin.test.Test
import omnia.util.test.fluent.Assertion.Companion.assertThat
import omnia.util.test.fluent.isA

class CachedDoubleTest {

  @Test
  fun cache_didReturnSimpleDoubleCache() {
    assertThat(CachedDouble.cache { 132.0 }).isA(SimpleDoubleCacher::class)
  }
}